### DISCUSSION TITLE
Embed video as background in a website

### DISCUSSION THREAD

**Discussion by AlexPreyer** [mediacms-io/mediacms#1009] [Q&A]
I have running Mediacms for some weeks now and for most things this great solution is what I need. But for one case I cannot find a solution. I host videos on my Mediacms server mainly for embed them on my websites (Wordpress with Elementor). In some cases, the videos are embeded as background videos. If I use there the Mediacms embed link, it doesn't work (video is not appearing). It only works when I use the direct links to the files with a certain resolution (e.g. "Download" - "1080p"). Embedding this links let run the video on the background of the website. But when doing so, the browser cannot select a suitable resolution for the screen size. Is there another link to embed videos?
Thanks in advance

**Comment by FTL1** [comment]
Is the player or the video actually being embedded [for the "working" version]?

Tldr:
Is it the mp4 [original, non-packaged] file or is it the master Playlist [which requires HLS to be supported]?

While mediacms knows what to do with HLS assets, WordPress doesn't..rather it needs to embed the player, not the video. How it does this handling of the player may be causing the problem.

Is the player being called, for the background play or is the video assets file being called directly [to get it to play]? 

Since the mediacms player is required for ABR [adaptive bitrate streaming, aka: multiple formats] your playback may actually be calling the original MP4, which will play automatically in any browser without the player.

As I avoid WordPress, i do not know, a WP savy person will hopefully speak to this. Perhaps a wp plugin/extension wrapper is needed?

Or the player just needs to be customized to remove all the controls and to autoplay?

Does MediaCMS have a WordPress plugin/extension?

**Comment by AlexPreyer** [comment]
First of all, thank you for your detailed answer, FTL1.
Unfortunately I don't know the answer to most of your questions.
What do you mean with "Is the player or the video actually being embedded [for the "working" version]?" - Do you mean in MediaCMS? Where can I do this.
Also this question: "Is it the mp4 [original, non-packaged] file or is it the master Playlist [which requires HLS to be supported]?" - I upload my mp4 files to MediaCMS. Then, with the "download" Button I select the link for the 1080 version. This link works in Wordpress. But I'm not sure if this is a good way or if this is the same as hosting the video file on my webserver.

Wordpress has no plugin for MediaCMS.

I'll ask in a Wordpress forum, as you said.

Thanks for your effort.
